src/components/side_panel/chart_panel.ts

```diff
@@ -178,10 +180,11 @@ export class ChartPanel extends Component<Props, SpreadsheetEnv> {
       this.state.fillColorTool = false;
       this.state.datasetDispatchResult = undefined;
       this.state.labelsDispatchResult = undefined;
+      this.state.chartSheetId = this.findSheetId(nextProps.figure.id);
       this.state.chart = this.env.getters.getChartDefinitionUI(
-        this.env.getters.getActiveSheetId(),
+        this.state.chartSheetId,
```

Comment:
why is it in the state ?

Reply:
no good reason , no need for reactivity as it changes with the figure - you're right